### PR TITLE
[audit log] Fix audit log persistence

### PR DESCRIPTION
**Pull request by geropl**
## Related Issue(s)

Fixes ENT-501

## How to test
 - call listWorkspaceSessions API like described [here](https://github.com/gitpod-io/gitpod/pull/19917#issuecomment-2186569397)
 - see how it does not throw an error in the server logs

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - gpl-fix-pa732bb07b4c</li>
	<li><b>🔗 URL</b> - <a href="https://gpl-fix-pa732bb07b4c.preview.gitpod-dev.com/workspaces" target="_blank">gpl-fix-pa732bb07b4c.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - gpl-fix-pagination-parsing-gha.27604</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-gpl-fix-pa732bb07b4c%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Review by mustard-mh — COMMENTED** [review]
I tried with preview env with actions: createWorkspace, createRepository, deleteRepository. Nothing recorded in database: 
- Preview's audit log feature is enabled
- No audit log printed in server
- No audit log persisted in database


image

**Comment by geropl** [comment]
> I tried with preview env with actions

:confused: Thank you @mustard-mh ! Let me have a look again what's going on here.

**Comment by mustard-mh** [comment]
@geropl I see, preview env needs to run with OIDC
```
        const user = await this.dbUser.findUserById(actorId);
        // audit logs are an enterprise feature, so we ignore user actions that are not associated with an organization
        if (!user || !user.organizationId) {
            return;
        }
```

**Comment by geropl** [comment]
@mustard-mh Exactly. Here's my link: https://gpl-fix-pa732bb07b4c.preview.gitpod-dev.com/login/gero's-org

Sorry for not mentioning earlier :bow:

**Comment by mustard-mh** [comment]
@geropl force updated our user with an organizationId. I can find auditLogs once, then found server error `TypeError: Do not know how to serialize a BigInt` again

**Comment by geropl** [comment]
> I can find auditLogs once, then found server error TypeError: Do not know how to serialize a BigInt again

Ah, PublicApiConverter. Let me fix it. Thx! :pray:

**Comment by geropl** [comment]
@mustard-mh Fixed the JSON.stringify in PublicApiConverter and deployed it, you should not see any errors anymore :+1:

File: components/gitpod-protocol/src/util/stringify.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+export namespace BigIntToJson {
+    export const BIGINT_KEY = "bigint:";
+    export function replacer(key: string, value: any) {
+        if (typeof value === "bigint") {
+            return `${BIGINT_KEY}${value.toString()}`;
```
Comment (mustard-mh) [inline components/gitpod-protocol/src/util/stringify.ts:11]: nit: string can be built by users easier than [object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json).

**Review by mustard-mh — APPROVED** [review]
List audit logs is working now 🚀 

image

File: components/gitpod-protocol/src/util/stringify.ts (hunk 1 above)
Comment (geropl) [reply components/gitpod-protocol/src/util/stringify.ts:11]: I know. I just don't like monkey-patching. :slightly_smiling_face:

**Pull request merged**